installer/install.iss

```diff
@@ -2797,6 +2804,52 @@ begin
     end;
 end;
 
+function UpgradeFromDotNetBasedScalar:Boolean;
+var
+    RegKey,UninstallScalar,ScalarExe,Cmd,OutPath,ErrPath:String;
+    Res:Longint;
+    Enlistments:TArrayOfString;
+    i:Integer;
+begin
+    Result:=True;
+
+    RegKey:='SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall\{82F731CB-1CFC-406D-8D84-8467BF6040C7}_is1';
+    if not RegQueryStringValue(HKEY_LOCAL_MACHINE,RegKey,'UninstallString',UninstallScalar) then
+        // No existing Scalar found; ignore this silently
+        Exit;
+
+    // Check twice that .NET-based Scalar is there
+    ScalarExe:=UninstallScalar;
+    StringChangeEx(ScalarExe,'"','',True); // strip the surrounding double-quote characters
+    StringChangeEx(ScalarExe,'\unins000.exe','\scalar.exe',True);
+    if not FileExists(ScalarExe) then
+        Exit;
+
+    // First, get .NET-based Scalar's idea of the registered enlistments
+    OutPath:=ExpandConstant('{tmp}\scalar-list.out');
+    ErrPath:=ExpandConstant('{tmp}\scalar-list.err');
+    Cmd:='/D /C ""'+ScalarExe+'" list  >"'+OutPath+'" 2>"'+ErrPath+'""';
+    if not Exec(ExpandConstant('{sys}\cmd.exe'),Cmd,'',SW_HIDE,ewWaitUntilTerminated,Res) or (Res<>0) then begin
+        LogError(ExpandConstant('Line {#__LINE__}: Unable to run `scalar list` (code: '+IntToStr(Res)+' output: '+ReadFileAsString(OutPath)+', errors: '+ReadFileAsString(ErrPath)+').'));
+        Result:=False;
+    end;
+    LoadStringsFromFile(OutPath,Enlistments);
+
+    // Now, register them with the C-based Scalar
+    for i:=0 to Length(Enlistments)-1 do begin
+        Cmd:='/D /C ""'+AppDir+'\cmd\scalar.exe" register "'+Enlistments[i]+'"  >"'+OutPath+'" 2>"'+ErrPath+'""';
+        if not Exec(ExpandConstant('{sys}\cmd.exe'),Cmd,'',SW_HIDE,ewWaitUntilTerminated,Res) or (Res<>0) then
+            LogError('Could not register "'+Enlistments[i]+'" with Scalar (code: '+IntToStr(Res)+' output: '+ReadFileAsString(OutPath)+', errors: '+ReadFileAsString(ErrPath)+'); ignoring');
+    end;
```

Comment:
Perhaps this should just be a "add these paths to the global `scalar.repo` config list" and then we want to run `scalar register` on each of those paths during every upgrade, anyway, to get the latest recommended config settings.

Reply:
Hmm. Maybe we can add a new verb for that? Or would it be something like `git for-each-repo --config=scalar.repos scalar register` (I'd have to teach `git for-each-repo` to special-case the argument `scalar` to not mean `git-scalar` but `scalar.exe` instead)?

Reply:
Right, I think I made note of that possible change to `git for-each-repo` to allow a non-git subcommand (`--no-git`?).

Reply:
We could take a page out of aliases' book and support `git for-each-repo --config=scalar.repos !scalar register`.

Reply:
I think I'd actually be a lot in favor of implementing `scalar upgrade-config [--all | -a]`.

Reply:
That works for me.

Reply:
Done.

Reply:
Wait, no, not here.

Reply:
I've decided to go with the verb name `reconfigure` (and make `config` a hard-coded alias).

Reply:
See https://github.com/microsoft/git/pull/359 for the implementation.